src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/geoMap.js

```diff
@@ -108,9 +108,13 @@ class GeoMapVisualization extends Visualization {
         this.setPreprocessorCode(`
                 df -> case df of
                     Table.Table _ ->
-                        columns = df.select ['label', 'latitude', 'longitude'] . columns
-                        serialized = columns.map (c -> ['df_' + c.name, c.to_vector])
-                        Json.from_pairs serialized . to_text
+                        names = ['label', 'latitude', 'longitude']
+                        look_for name = 
+                            is_matching column = column.name.equals_ignore_case name
+                            ["df_" + name, df.columns.find is_matching . to_vector]
+                        valid_column pair = pair.at 1 . is_error . not
```

Comment:
does it work with empty dataframe? If so, I don't see any warnings, but also I was not digging deeply in it

Reply:
@wdanilo 
It will return an empty JSON which is the behavior that was present so-far.
Not sure if this should be considered working but even if not — it should be a separate issue.